### PR TITLE
Adds Constants for the Scheme and Bearer format types

### LINKED ISSUE #632: Consider adding constants for scheme and bearer
Consider adding constants for various schemes and bearer formats.

From:
```cs
c.AddSecurityDefinition("bearerAuth", new OpenApiSecurityScheme
{
    Type = SecuritySchemeType.Http,
    Scheme = "bearer", // string
    BearerFormat = "JWT", // string
    Description = "JWT Authorization header using the Bearer scheme."
});
```

To:
```cs
c.AddSecurityDefinition("bearerAuth", new OpenApiSecurityScheme
{
    Type = SecuritySchemeType.Http,
    Scheme = Scheme.Bearer, // constant
    BearerFormat = BearerFormat.Jwt, // constant
    Description = "JWT Authorization header using the Bearer scheme."
});
```

### PR DESCRIPTION
**Pull request by MaggieKimani1**
Fixes #632

### CODE REVIEW

File: src/Microsoft.OpenApi/Models/AuthenticationScheme.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+// Licensed under the MIT license. 
+
+using Microsoft.OpenApi.Attributes;
+
+namespace Microsoft.OpenApi.Models
+{
+    /// <summary>
+    /// The type of HTTP authentication schemes
+    /// </summary>
+    public enum AuthenticationScheme
```
Comment (darrelmiller) [inline src/Microsoft.OpenApi/Models/AuthenticationScheme.cs:11]: This is a thing of beauty.  However, we can't do it this way.  The scheme property is defined as a string in the spec https://spec.openapis.org/oas/v3.0.0#fixed-fields-22 and the [IANA HTTP Auth Scheme](https://www.iana.org/assignments/http-authschemes/http-authschemes.xhtml) registry defines the allow values.  That registry can have new values added at any time.
Also, changing a public member from string to enum is a breaking change, so we couldn't do it in 1.3.

File: src/Microsoft.OpenApi/Models/AuthenticationScheme.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+        /// <summary>
+        /// Use bearer Authentication scheme
+        /// </summary>
+        [Display("bearer")] Bearer,
+
+        /// <summary>
+        /// Use OpenIdConnectUrl
+        /// </summary>
+        [Display("openIdConnectUrl")] OpenIdConnectUrl
```
Comment (darrelmiller) [inline src/Microsoft.OpenApi/Models/AuthenticationScheme.cs:26]: This is an odd one.  This isn't an HTTP auth scheme.    This is a property on the OpenAPI Security Scheme object.

File: src/Microsoft.OpenApi/Models/AuthenticationScheme.cs (hunk 1 above)
Comment (darrelmiller) [reply src/Microsoft.OpenApi/Models/AuthenticationScheme.cs:11]: I think it should be sufficient to add Bearer and Basic as constants in this file https://github.com/microsoft/OpenAPI.NET/blob/649f0b1780c575c4780f4091eedd7a555e8ccc4a/src/Microsoft.OpenApi/Models/OpenApiConstants.cs

**Review by darrelmiller — CHANGES_REQUESTED** [review]
Replace enums with consts

File: src/Microsoft.OpenApi/Models/AuthenticationScheme.cs (hunk 1 above)
Comment (MaggieKimani1) [reply src/Microsoft.OpenApi/Models/AuthenticationScheme.cs:11]: Thank you for pointing this out. Resolving..

File: test/Microsoft.OpenApi.Tests/Models/OpenApiComponentsTests.cs [hunk 3]
```diff
@@ -57,7 +57,7 @@ public class OpenApiComponentsTests
                 {
                     Description = "description1",
                     Type = SecuritySchemeType.OpenIdConnect,
-                    Scheme = "openIdConnectUrl",
+                    Scheme = OpenApiConstants.OpenIdConnectUrl,
```
Comment (darrelmiller) [inline test/Microsoft.OpenApi.Tests/Models/OpenApiComponentsTests.cs:60]: Remove this constant. It isn't required.

Comment (darrelmiller) [reply test/Microsoft.OpenApi.Tests/Models/OpenApiComponentsTests.cs:60]: Set the scheme to be bearer

**Review by darrelmiller — APPROVED** [review]

**Pull request merged**